### ISSUE TITLE
Different configurations for the same property in the base and child classes for Table-per-concrete-type (TPC)

### ISSUE THREAD

**Issue by StakhurA** [dotnet/efcore#30520]
## Can we have different configurations for the same property in both the base and child classes while using the Table-per-Concrete-Type (TPC) strategy?

**Example:**
My intention is to use a regular identity (database-generated) column for SequenceNumber property in Blog entity, and configure the SequenceNumber property in ArchivedBlog entity in a manner that it will not function as an identity column generated by the database.

```c#
    public class ApplicationDbContext : DbContext
    {
        public DbSet<Blog> Blogs { get; set; }
        public DbSet<ArchivedBlog> ArchivedBlog { get; set; }


        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            var blogBuilder = modelBuilder.Entity<Blog>()
                .UseTpcMappingStrategy();
            blogBuilder.Property<long>("SequenceNumber")
                .ValueGeneratedOnAdd();

            var archivedBlogBuilder = modelBuilder.Entity<ArchivedBlog>();
            archivedBlogBuilder.Property<long>("SequenceNumber");
        }
    }

    public class Blog
    {
        public Guid BlogId { get; set; }
        public string Url { get; set; }
    }

    public class ArchivedBlog : Blog
    {
    }
``` 
However, as it is shown in the generated migration below, if I use TpcMappingStrategy for the Blog entity and configure SequenceNumber property as an identity column in Blog entity (parent), then in ArchivedBlog entity (child) it will be configured in the same way, even if I don't configure SequenceNumber as an identity column there. As result, it creates a single sequence for both tables, which is used to guarantee SequenceNumber uniqueness in a hierarchy. 

So, it creates the following migration:
![image](https://user-images.githubusercontent.com/10417205/225933503-b788f77a-82e5-44dc-bb42-0e6c0eea37c4.png)

Basically, my intention is to have the following migration for class hierarchy from the example:
![image](https://user-images.githubusercontent.com/10417205/225933011-5f119e78-4215-4619-80c4-87cc83182b88.png)
As you can see only SequenceNumber property in Blog entity is configured as an identity column.
(it can be achieved by introducing a base class for example BlogBase with all Blog's properties and inheriting this base class by Blog and ArchivedBlog classes. However, I would like to have a class hierarchy which is described in the example)

EF Core version: Microsoft.EntityFrameworkCore Version=7.0.3
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework:  .NET 7.0

**Comment by StakhurA** [comment]
@ajcvickers thanks for the reply. yes, your summary is correct.
Inheritance is used here because we need to copy data from the table mapped to the base class to the table which is mapped to the child class, so all properties/columns that are in the base class should remain identical to those in the child class.

**Comment by ajcvickers** [comment]
@StakhurA This isn't something we are going to implement, since it seems like a quite tenuous use of inheritance. However, if I understand your use case correctly, then every ArchivedBlog created will already have a sequence number set, so the default value constraint will never be used in this case. Beyond this, if you want to ensure it isn't used accidentally, then you can edit the migration to remove the constraint for the ArchivedBlogs table.
